casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -87,12 +57,60 @@ object InterpreterUtil {
     }
   }
 
-  private def computeParentsPostState(parents: Seq[BlockMessage],
-                                      genesis: BlockMessage,
-                                      dag: BlockDag,
-                                      defaultStateHash: StateHash,
-                                      knownStateHashes: Set[StateHash],
-                                      runtimeManager: RuntimeManager)(
+  def computeDeploysCheckpoint(
+      parents: Seq[BlockMessage],
+      deploys: Seq[Deploy],
+      genesis: BlockMessage,
+      dag: BlockDag,
+      defaultStateHash: StateHash,
+      knownStateHashes: Set[StateHash],
```

Comment:
I wonder if we should abandon `StateHash` in favour of `Checkpoint` entirely. It would make the return type of all these various `compute*` functions consistent.

Reply:
Isn't it expensive to keep track of trace logs?

Reply:
Yes, perhaps you are right. Ok, perhaps it is fine as is.